src/api-linux.c

```diff
@@ -941,3 +944,32 @@ SEXP psll_open_files(SEXP p) {
   UNPROTECT(1);
   return result;
 }
+
+SEXP ps__users() {
+  struct utmp *ut;
+  SEXP result;
+  PROTECT_INDEX pidx;
+  int len = 10, num = 0;
+
+  PROTECT_WITH_INDEX(result = allocVector(VECSXP, len), &pidx);
+
+  setutent();
+
+  while ((ut = getutent()) != NULL) {
+
+    if (ut->ut_type != USER_PROCESS) continue;
+
+    if (++num == len) {
+      len *= 2;
+      REPROTECT(result = Rf_lengthgets(result, len), pidx);
+    }
+    SET_VECTOR_ELT(
+      result, num,
+      ps__build_list("sssdi", ut->ut_user, ut->ut_line, ut->ut_host,
+		     (double) PS__TV2DOUBLE(ut->ut_tv), ut->ut_pid));
+  }
+
+  endutent();
```

Comment:
Don't you need to resize the vector to the actual number of users found at the end here? Right now it is over-allocated I think?

Reply:
There is a `not_null()` call in R. That is always needed on some platforms, and removes the trailing  `NULL`s as well

Reply:
oh I see.